File: 1-serverless-ts/cdk/config-tracker.app.ts
#!/usr/bin/env node
import * as cdk from 'aws-cdk-lib';
import { Duration } from 'aws-cdk-lib';
import { Architecture, Runtime } from 'aws-cdk-lib/aws-lambda';
import { NodejsFunctionProps } from 'aws-cdk-lib/aws-lambda-nodejs';
import { RetentionDays } from 'aws-cdk-lib/aws-logs';
import { ConfigTrackerStack } from './config-tracker.stack';

export const resourcePrefix = 'config-tracker-1-serverless-ts';

export const baseLambdaProps: NodejsFunctionProps = {
  runtime: Runtime.NODEJS_22_X,
  architecture: Architecture.ARM_64,
  memorySize: 512,
  timeout: Duration.seconds(10),
  logRetention: RetentionDays.ONE_WEEK,
  bundling: { minify: true, sourceMap: true, externalModules: [
    '@aws-sdk/client-dynamodb',
    '@aws-sdk/client-sns',
    '@aws-sdk/lib-dynamodb',
    '@aws-sdk/util-dynamodb'
  ]},
};

const configTrackerApp = new cdk.App();
new ConfigTrackerStack(configTrackerApp, `${resourcePrefix}-stack`, {});

File: 2-serverless-java/src/main/java/com/github/moravcik/configtracker/cdk/ConfigTrackerApp.java
package com.github.moravcik.configtracker.cdk;

import software.amazon.awscdk.App;
import software.amazon.awscdk.StackProps;

import java.util.Arrays;

public class ConfigTrackerApp {
    public static String resourcePrefix = "config-tracker-2-serverless-java";

    public static void main(final String[] args) {
        App app = new App();
        new ConfigTrackerStack(app, resourcePrefix + "-stack", StackProps.builder().build());
        app.synth();
    }
}

